Dockerfile

```
@@ -1,26 +1,30 @@
-ARG OS_VER
-FROM --platform=linux/amd64 nrel/openstudio:$OS_VER as buildstockbatch
-ARG CLOUD_PLATFORM=aws
-ENV DEBIAN_FRONTEND=noninteractive
+FROM nrel/openstudio:3.8.0
+
+# Set environment variables
+ENV LANG=C.UTF-8 LC_ALL=C.UTF-8
+ARG DEBIAN_FRONTEND=noninteractive
+
+# Install Python 3.11 using deadsnakes PPA and necessary tools
+RUN apt-get update && apt-get install -y --no-install-recommends \
+    software-properties-common wget build-essential libssl-dev libffi-dev zlib1g-dev \
+    && add-apt-repository ppa:deadsnakes/ppa \
+    && apt-get update \
+    && apt-get install -y --no-install-recommends python3.11 python3.11-venv python3.11-dev \
+    && apt-get clean \
+    && rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*
+
+# Set up virtual environment
+ENV VENV_DIR=/opt/venv
+RUN python3.11 -m venv $VENV_DIR
+ENV PATH=$VENV_DIR/bin:$PATH
+
+# Install Python dependencies
 COPY . /buildstock-batch/
-COPY nrel_root_ca.crt /usr/local/share/ca-certificates/
-RUN update-ca-certificates
+RUN pip install --no-cache-dir --upgrade pip && \
```

I think we still want to keep the nrel certificate thing to make it work inside NREL VPN.
```suggestion
COPY nrel_root_ca.crt /usr/local/share/ca-certificates/
RUN update-ca-certificates
RUN pip install --no-cache-dir --upgrade pip && \
```